### PR TITLE
fix(react-select): FilterBox focus event and adhoc filter popup height

### PR DESCRIPTION
**Pull request by ktmud**
### SUMMARY

This fixes two bugs related to [recent migration of react-select](https://github.com/apache/incubator-superset/pull/9628).

1. Focusing on FilterBox select should open the filter indicators:
   image

2. Select a lot of filter values in AdhocFilter editor should expand filter popup height.

   image

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

See above.

### TEST PLAN

Added a Cypress test case for the first bug.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/cypress-base/cypress/integration/dashboard/filter.js [hunk 1]
```diff
@@ -56,8 +56,26 @@ export default () =>
...
         .click({ force: true });
+
+      // should open the filter indicator
+      cy.get('.filter-indicator.active')
+        .should('be.visible')
+        .should($node => {
+          expect($node).to.have.length(9);
+        });
+
+      cy.get('.chart-header').first().click({ force: true });
```
Comment (ktmud) [inline superset-frontend/cypress-base/cypress/integration/dashboard/filter.js:67]: This is for triggering `blur` on the filter select.

File: superset-frontend/cypress-base/cypress/integration/dashboard/filter.js [hunk 2]
```diff
@@ -56,8 +56,26 @@ export default () =>
       cy.get('.Select__control')
         .contains('Select [region]')
         .click({ force: true });
+
+      // should open the filter indicator
+      cy.get('.filter-indicator.active')
+        .should('be.visible')
+        .should($node => {
```
Comment (etr2460) [inline superset-frontend/cypress-base/cypress/integration/dashboard/filter.js:63]: why prefix node with a dollar sign? Can we get rid of that in the variable name?

File: superset-frontend/cypress-base/cypress/integration/dashboard/filter.js [hunk 3]
```diff
@@ -56,8 +56,26 @@ export default () =>
...
+          expect($node).to.have.length(9);
+        });
+
+      cy.get('.chart-header').first().click({ force: true });
+
+      // should hide the filter indicator
+      cy.get('.filter-indicator')
+        .not('.active')
+        .should($node => {
```
Comment (etr2460) [inline superset-frontend/cypress-base/cypress/integration/dashboard/filter.js:72]: same comment about the dollar sign here

**Review by graceguo-supercat — APPROVED** [review]
LGTM

**Review by etr2460 — COMMENTED** [review]
a couple comments, otherwise looks good

File: superset-frontend/cypress-base/cypress/integration/dashboard/filter.js (hunk 2 above)
Comment (ktmud) [reply superset-frontend/cypress-base/cypress/integration/dashboard/filter.js]: Was just following Cypress's official doc, but I agree it's probably not necessary.

**Review by etr2460 — APPROVED** [review]

**Pull request merged**